metadata/moa/scripts/nbconverted/2.map-broad_id.py

```diff
@@ -0,0 +1,117 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# In[1]:
+
+
+import pandas as pd
+import numpy as np
+import rdkit.Chem.inchi as inchi
+
+
+# Sample files from each version of the repurposing hub are read
+
+# In[2]:
+
+
+samples_2017 = pd.read_csv('clue/repurposing_samples_20170327.txt',usecols=['broad_id','pert_iname','InChIKey'],delimiter='\t', comment='!', encoding='iso-8859-1')
+samples_2018a = pd.read_csv('clue/repurposing_samples_20180516.txt', usecols=['broad_id','pert_iname','InChIKey','deprecated_broad_id'], delimiter='\t', comment='!', encoding='iso-8859-1')
+samples_2018b = pd.read_csv('clue/repurposing_samples_20180907.txt', usecols=['broad_id','pert_iname','InChIKey','deprecated_broad_id'], delimiter='\t', comment='!', encoding='iso-8859-1')
+samples_2020 = pd.read_csv('clue/repurposing_samples_20200324.txt', usecols=['broad_id','pert_iname','InChIKey','deprecated_broad_id'], delimiter='\t', comment='!', encoding='iso-8859-1')
+
+
+# In[3]:
+
+
+# 2017 version is missing deprecated_broad_id
+samples_2017['deprecated_broad_id'] = np.nan
+
+
+# In[4]:
+
+
+# The first 13 characters of the Broad ID and the first 14 characters of InChIKey are extracted
+# Year names are appended to column names
+# For some rows InChI are extracted instead of InChIKey -> these are fixed
+
+def id_cleanup(df, year):
+    df.dropna(subset=['InChIKey'], inplace=True)
```

Comment:
you're going to love [chaining in pandas](https://towardsdatascience.com/the-unreasonable-effectiveness-of-method-chaining-in-pandas-15c2109e3c69)! This is similar to chaining in `dplyr`.

Many of the subsequent lines won't work with chaining, so it is ok to split them up.

Reply:
Thanks for suggestion! I have been meaning to learn how to do it but now I got the chance to force myself to learn :)